org.gecko.weather.api/src/org/gecko/weather/api/util/DWDUtils.java

```diff
@@ -433,176 +436,163 @@ public static void setMOSMIXMeasurement(MOSMIXSWeatherReport report, String meas
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_EFFECTIVE;
 			break;
 		case "Nh":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_HIGH;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_HIGH;
 			break;
 		case "Nm":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_MID;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_MID;
 			break;
 		case "Nl":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_LOW;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_LOW;
 			break;
 		case "N05":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_BELOW500;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__CLOUD_COVER_BELOW500;
 			break;
 		case "VV":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__VISIBILITY;
-			// castedValue = Integer.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__VISIBILITY;
 			break;
 		case "wwM":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST1;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST1;
 			break;
 		case "wwM6":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST6;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST6;
 			break;
 		case "wwMh":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST12;
-			// castedValue = Short.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__FOG_PROP_LAST12;
 			break;
 		case "ww":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__SIGNIFICANT_WEATHER;
-			// castedValue = Integer.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__SIGNIFICANT_WEATHER3_HOURS;
+			int wwCode = Math.round((float) value);  // e.g., 1.0 → 1
+			String wwEnumKey = WMOWeatherCodeType.WUNKNOWN.getLiteral();
+			if(wwCode >= 0 && wwCode <= 99) {
+				wwEnumKey = String.format("%02d", wwCode); // → "01"
+			}
+			WMOWeatherCodeType wwEnumCode = WMOWeatherCodeType.get(wwEnumKey);
+			if(wwEnumCode == null) wwEnumCode = WMOWeatherCodeType.WUNKNOWN;
+			value = wwEnumCode;
 			break;
 		case "W1W2":
-			attr = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__PAST_WEATHER;
-			// castedValue = Integer.valueOf(value.toString());
+			feature = WeatherPackage.Literals.MOSMIXS_WEATHER_REPORT__SIGNIFICANT_WEATHER6_HOURS;
+			int code = Math.round((float) value); // e.g. → 30.0 -> 30
```

Comment:
EMF generates a method get(int) for Enumsto get literals. It seems like you are doing String magic here to call the get(String) method at the enum. Maybe it easier to use the Literal#value (int) to get the right literal?